agent/agent_test.go

```diff
@@ -93,7 +93,7 @@ func TestAgent_ConnectClusterIDConfig(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			a := &TestAgent{Name: tt.name, HCL: tt.hcl}
+			a := &TestAgent{Name: tt.name, HCL: tt.hcl, LogOutput: testutil.TestWriter(t)}
```

Comment:
This caught my attention because it is not using the constructor. But logs are going to the correct place nevertheless.

Reply:
Just saw that failures are expected and thus the constructor cannot be used, right?

Reply:
yea! the thing with this test case is that it tests that `a.Start()` returns an error. But since the constructors have retry logic for when `a.Start()` returns errors, in this case it would retry unnecessarily a bunch of times before returning the error. So that's why I left this one case out.

I'll add a comment. That should be more clear. Thx for noting!